arraylist.c

```diff
@@ -22,7 +22,7 @@
 #include "common.h"
 
 /* arraylist implementation */
-void arraylist_init(struct arraylist_t *arr, u32 T_size)
+void arraylist_init(struct arraylist_t *arr, size_t T_size)
```

Comment:
I disagree T_size should be size_t. size_t is used to store the size of an object, true, but you will never have a single object of that size. size_t is a good choice for storing the size of a collection since it guarantees it is enough to contain the size of the entire collection

Reply:
Size_t is still the C standard for sizes however.